java-squid/src/main/java/org/sonar/java/se/SymbolicValue.java

```diff
@@ -286,9 +278,7 @@ public InstanceOfSymbolicValue(int id) {
         List<ProgramState> ps = operand.setConstraint(programState, NullConstraint.NOT_NULL);
         if (ps.size() == 1 && ps.get(0).equals(programState)) {
           //FIXME we already know that operand is NOT NULL, so we add a different constraint to distinguish program state. Typed Constraint should store the deduced type.
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1134)